core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -634,25 +650,29 @@ protected Object visitInPredicate(InPredicate node, Object context)
             boolean hasUnresolvedValue = value instanceof Expression;
             boolean hasNullValue = false;
             boolean found = false;
+
+            List<Object> inListValues;
+            if (canUseCache(context)) {
+                inListValues = (List<Object>) optimizationCache.computeIfAbsent(valueList, key -> processInListExpression(value, valueList, context));
```

Comment:
Seems like invoking `optimizationCache` here is an unnecessary complexity.
I would assume caching the `process` calls is what actually matters.

Reply:
You are right, I ran benchmarks but there are no major differences.

Even for queries like the below, no major differences

```
largeInQuery = "SELECT * from orders where o_orderkey in " +
                    IntStream.range(0, 100000)
                    .mapToObj(Integer::toString)
                    .collect(joining(", ", "(", ")")) + "AND o_custkey in " +
                    IntStream.range(0, 100000)
                    .mapToObj(Integer::toString)
                    .collect(joining(", ", "(", ")"));
```